DatacenterIntegration/Portal/PortalUserDataGdprUtilities.psm1

```diff
@@ -0,0 +1,310 @@
+﻿<###################################################
+ #                                                 #
+ #  Copyright (c) Microsoft. All rights reserved.  #
+ #                                                 #
+ ##################################################>
+
+$ErrorActionPreference = "Stop"
+
+<#
+.Synopsis
+   Clear the portal user data under AAD environment
+#>
+function Clear-AzsUserDataUnderAAD
+{
+    param
+    (
+        # The user credential with which to acquire an access token targeting Graph.
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNull()]
+        [pscredential] $AadGraphCred = $null,
```

Comment:
This will not work with MFA enabled; check the multitenancy scripts for a pattern to use, where we use Azure PS to handle the authentication and then grab the refresh token from that.

Reply:
Looks like you are doing that; make the credential optional and only pass it to add-azurermaccount if it is provided; this allows MFA to work by allowing that cmdlet to pop-up the login flow, as well as with a credential

Reply:
Thanks.